stm32_base/hal/imu_stm32/invensense_9150.c

```diff
@@ -120,6 +290,46 @@ int is9150_read_accel(struct is9150_accel_data *data)
     return res;
 }
 
+int is9150_read_mag(struct is9150_mag_data *data)
+{
+    uint8_t reg_res[6] = {0};
+    int res = is9150_read_reg_block(IS_REG_EXT_SENS_DATA_00, IS_MAG_MEAS_COUNT,
+                                    reg_res);
+
+    data->mag_x = (((int16_t)reg_res[1]) << 8) | reg_res[0];
+    data->mag_y = (((int16_t)reg_res[3]) << 8) | reg_res[2];
+    data->mag_z = (((int16_t)reg_res[5]) << 8) | reg_res[4];
+    if (TRACE_LEVEL) {
+        pr_trace("mag raw: ");
+        pr_trace_int(data->mag_x);
+        pr_trace(" ");
+        pr_trace_int(data->mag_y);
+        pr_trace(" ");
+        pr_trace_int(data->mag_z);
+        pr_trace(" ");
+        pr_trace("\r\n");
+    }
+    data->mag_x = ((int32_t)data->mag_x * mag_sens_adj[0]);
+    data->mag_y = ((int32_t)data->mag_y * mag_sens_adj[1]);
+    data->mag_z = ((int32_t)data->mag_z * mag_sens_adj[2]);
+    if (TRACE_LEVEL) {
+        pr_trace("mag raw adj: ");
+        pr_trace_int(data->mag_x);
+        pr_trace(" ");
+        pr_trace_int(data->mag_y);
+        pr_trace(" ");
+        pr_trace_int(data->mag_z);
+        pr_trace(" ");
+        pr_trace("\r\n");
+    }
+    float heading = atan2((double)data->mag_y, (double)data->mag_x) * 180.0/3.14159265 + 180;
```

Comment:
atan2 can only return values between -PI and PI right?  So you can clean this up no?  Look at my GPS bearing code for example.